service/frontend/nexus_http_response_writer.go

```diff
@@ -0,0 +1,67 @@
+// The MIT License
+//
+// Copyright (c) 2024 Temporal Technologies Inc.  All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package frontend
+
+import (
+	"net/http"
+)
+
+const (
+	// The Failure-Source header is used to indicate from where the Nexus failure originated.
+	failureSourceHeaderName = "Failure-Source"
+	// failureSourceServer indicates the failure originated within the Nexus server.
+	failureSourceServer = "SERVER"
+	// failureSourceClient indicates the failure originated from outside the server (e.g. bad request or on the Nexus worker).
+	failureSourceClient = "CLIENT"
```

Comment:
```suggestion
	// failureSourceServer indicates the failure originated within the Temporal server.
	failureSourceServer = "server"
	// failureSourceClient indicates the failure originated from a user worker.
	failureSourceWorker = "worker"
```

Reply:
Thinking about this some more, just drop the "server" source here, we should just assume it's the server if empty so we don't have to make sure to set this header for every failure path.